voluptuous/humanize.py

```diff
@@ -0,0 +1,39 @@
+from voluptuous import Invalid, MultipleInvalid
+from voluptuous.error import Error
+
+
+MAX_VALIDATION_ERROR_ITEM_LENGTH = 500
+
+
+def _nested_getitem(data, path):
+    for item_name in path:
+        try:
+            data = data[item_name]
+        except KeyError:
+            return None
+    return data
+
+
+def humanize_error(data, validation_error, max_sub_error_length=MAX_VALIDATION_ERROR_ITEM_LENGTH):
+    """ Provide a more helpful + complete validation error message than that provided automatically
+    Invalid and MultipleInvalid do not include the offending value in error messages,
+    and MultipleInvalid.__str__ only provides the first error.
+    """
+    if isinstance(validation_error, MultipleInvalid):
+        return '\n'.join(sorted(
+            humanize_error(data, sub_error, max_sub_error_length)
+            for sub_error in validation_error.errors
+        ))
+    else:
+        offending_item = _nested_getitem(data, validation_error.path)
+        offending_item_summary = repr(offending_item)
```

Comment:
or are you looking for a `partial` + `map` instead of a generator expression?
I didn't think there was a need to deal with nested MultipleInvalids